api/src/DataPersister/InvitationDataPersister.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace App\DataPersister;
+
+use ApiPlatform\Core\DataPersister\ContextAwareDataPersisterInterface;
+use ApiPlatform\Core\Exception\ResourceClassNotSupportedException;
+use App\DTO\Invitation;
+use App\Entity\CampCollaboration;
+use App\Repository\CampCollaborationRepository;
+use App\Repository\UserRepository;
+use App\Util\ValidationExceptionFactory;
+use Symfony\Component\Security\Core\Security;
+
+class InvitationDataPersister implements ContextAwareDataPersisterInterface {
+    public function __construct(
+        private ContextAwareDataPersisterInterface $dataPersister,
+        private CampCollaborationRepository $campCollaborationRepository,
+        private Security $security,
+        private UserRepository $userRepository
+    ) {
+    }
+
+    public function supports($data, array $context = []): bool {
+        return $data instanceof Invitation;
+    }
+
+    /**
+     * @param Invitation $data
+     *
+     * @throws ResourceClassNotSupportedException
+     *
+     * @return object|void
+     */
+    public function persist($data, array $context = []) {
+        if (Invitation::ACCEPT === ($context['item_operation_name'] ?? null)) {
+            if ($data->userAlreadyInCamp) {
+                $message = "The user {$data->userDisplayName} is already associated with the camp {$data->campTitle}";
+
+                throw ValidationExceptionFactory::createValidationException($message);
+            }
```

Comment:
Kannst du das stattdessen mit einem Validator auf `userAlreadyInCamp` machen der nur in der `accept` Operation läuft? Dazu `'validation_groups' => ['Default', 'accept']` auf der Operation deklarieren, und z.B. den Validator `#[Assert\IsFalse(groups: ['accept'], message: 'This user is already associated with the camp.')]` verwenden.

Reply:
Danke für die Idee, werde ich versuchen.

Reply:
Done